src/libraries/System.IO.FileSystem/tests/RandomAccess/SectorAlignedMemory.Windows.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using static Interop.Kernel32;
+
+namespace System.IO.Tests
+{
+    internal class SectorAlignedMemory<T> : MemoryManager<T>
+    {
+        private bool disposed = false;
+        private int refCount = 0;
+        private IntPtr memory;
+        private int length;
+
+        private unsafe SectorAlignedMemory(void* memory, int length)
+        {
+            this.memory = (IntPtr)memory;
+            this.length = length;
+        }
+
+        ~SectorAlignedMemory()
```

Comment:
This is generally considered a no-no.  Consider code like:
```C#
Foo(new SectorAlignedMemory<int>.Allocate(42).Memory.Span);
...
private void Foo(Span<int> span) { ... }
```
There's nothing in the span that references the SectorAlignedMemory, and it could become eligible for GC and finalization while the span is still being used, such that the app is then reading/writing deallocated memory.

cc: @GrabYourPitchforks

Reply:
In a non-shipping unit test project I'm a little less concerned. But Steve's right: we don't want this pattern to propagate through the framework.

In the `BoundedMemory` code (see https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/TestUtilities/System/Buffers/BoundedMemory.Windows.cs), we don't have a finalizer, and we rely on all callers to use the pattern `using var boundedMem = BoundedMemory.Create(...);`. Then it'll be properly destroyed at the end of the unit test.